### PR TITLE
display data 

### PR DESCRIPTION
**Pull request by SarahWett**
**Whats new:**

- displayed All Entries page 
- by clicking the all entries page you can see all related entries with date 

**comments**

- after changing the code logic I have to re-construct the form data reset (doesn't work out at the moment) 

[US Here ](https://github.com/SarahWett/capstone-project/issues/11)

### CODE REVIEW

File: components/Entry/index.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import styled from "styled-components";
+
+const StyledEntry = styled.article`
+  display: flex;
+  flex-direction: column;
+  gap: 30px;
+  padding: 16vw;
+  width: 80vw;
+  /* From https://css.glass */
```
Comment (stefankorder) [inline components/Entry/index.jsx:9]: Please delete this out commented code.

File: components/Entry/index.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+`;
+
+const StyledDate = styled.time`
+  display: flex;
+  justify-content: center;
+  text-transform: uppercase;
+  align-items: center;
+  font-size: 14px;
+  font-weight: 600;
+  /* From https://css.glass */
```
Comment (stefankorder) [inline components/Entry/index.jsx:24]: Please delete this out commented code.

File: components/Entry/index.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+`;
+
+const StyledTags = styled.div`
+  display: flex;
+  justify-content: center;
+  text-transform: uppercase;
+  align-items: center;
+  font-size: 14px;
+  font-weight: 600;
+  /* From https://css.glass */
```
Comment (stefankorder) [inline components/Entry/index.jsx:66]: Please delete this out commented code.

File: components/Entry/index.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  id,
+  formData,
+  setFormData,
+  entry,
+}) {
+  const keys = tags ? Object.keys(tags) : [];
+  const filtered = keys.filter((key) => {
+    return tags[key];
+  });
+  console.log(filtered);
```
Comment (stefankorder) [inline components/Entry/index.jsx:91]: Please delete this `console.log()` if it is just for debugging.

File: components/Form/index.js [hunk 5]
```diff
@@ -91,6 +81,12 @@ export default function Form({}) {
           <Tags
             formData={formData}
             setFormData={setFormData}
+            entries={entries}
+            setEntries={setEntries}
+            onAddEntry={handleAddEntry}
+            // setCheckedCheckbox={setCheckedCheckbox}
+            // checkedCheckbox={checkedCheckbox}
+            // isChecked={checkedCheckbox.isChecked}
```
Comment (stefankorder) [inline components/Form/index.js:89]: Please delete this out commented code.

File: components/Form/index.js [hunk 6]
```diff
@@ -159,22 +161,34 @@ export default function Form({}) {
...
+      //       hobby: false,
+      //       household: false,
+      //       tv: false,
+      //       sports: false,
+      //       walk: false,
+      //     },
+
+      //     message: "",
+      //   },
+      // ]);
```
Comment (stefankorder) [inline components/Form/index.js:187]: Please delete this out commented code.

File: components/QuestionOne/index.js [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+  listOfOptions,
+  entries,
+  smiley,
+  setEntries,
+}) {
   function handleOnChange(event) {
-    setFormData({ ...formData, smiley: event.target.value });
-    // setIsSmileySelected(true);
+    setEntries({ smiley: event.target.value });
+    // setFormData(...formData, { ...entries });
```
Comment (stefankorder) [inline components/QuestionOne/index.js:11]: Please delete this out commented code.

File: components/QuestionThree/index.js [hunk 8]
```diff
@@ -4,15 +4,17 @@ const StyledTextarea = styled.textarea`
   width: 100%;
 `;
 
-const Message = ({ formData, setFormData }) => {
+const Message = ({ formData, setFormData, entries, setEntries }) => {
   function handleOnChange(event) {
-    setFormData({ ...formData, message: event.target.value });
+    setEntries({ ...entries, message: event.target.value });
+
+    // setFormData({ ...formData, message: event.target.value });
```
Comment (stefankorder) [inline components/QuestionThree/index.js:11]: Please delete this out commented code.

File: pages/_app.js [hunk 9]
```diff
@@ -1,14 +1,57 @@
...
+  });
+
+  const [entries, setEntries] = useState([]);
+
+  function handleAddEntry(newEntry) {
+    const date = new Date().toLocaleDateString("en-us", {
+      dateStyle: "medium",
+    });
+
+    // const newFormData = ;
```
Comment (stefankorder) [inline pages/_app.js:36]: Please delete this out commented code.

File: pages/dashboard/AllEntries/index.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+          All Entries{" "}
+          <Badge isActive={filter === "all"}>{allEntriesCount}</Badge>
+        </StyledBatch>
+        <StyledEntries>
+          {formData?.map((data) => (
+            <Fragment key={data.id}>
+              {/* {index > 0 ? <Divider /> : null} */}
+              {/* {formData
+  .sort((a, b) => new Date(b.date) - new Date(a.date))
+  .map((data) => ( */}
```
Comment (stefankorder) [inline pages/dashboard/AllEntries/index.js:66]: Please delete this out commented code.

File: pages/dashboard/AllEntries/index.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+              <Entry
+                key={data.id}
+                smiley={data.smiley}
+                message={data.message}
+                date={data.date}
+                tags={data.tags}
+                id={data.id}
+                // formData={formData}
+                // setFormData={setFormData}
+                // data={data}
```
Comment (stefankorder) [inline pages/dashboard/AllEntries/index.js:76]: Please delete this out commented code.

File: pages/dashboard/AllEntries/index.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+              />
+              {/* ))} */}
```
Comment (stefankorder) [inline pages/dashboard/AllEntries/index.js:78]: Please delete this out commented code.

File: pages/_app.js [hunk 13]
```diff
@@ -1,14 +1,57 @@
...
+    setFormData([...formData, { ...entries, date }]);
+  }
+  console.log(formData);
```
Comment (stefankorder) [inline pages/_app.js:39]: Please delete this `console.log()` if it is just for debugging.

File: components/Form/index.js [hunk 14]
```diff
@@ -129,7 +125,13 @@ export default function Form({}) {
       return (
         <>
           {" "}
```
Comment (stefankorder) [inline components/Form/index.js:127]: Delete this empty space if possible.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Looks good so far. Take a look at my comments.

File: components/Entry/index.jsx [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+          <h2>
+            <StyledSmileySection>{smiley}</StyledSmileySection>
+          </h2>
+        </StyledEntrySmileyContainer>
+        <StyledNotes>{message}</StyledNotes>
+        {tags.map((key) => (
+          <StyledTags key={key}>{key}</StyledTags>
+        ))}
+
+        <div></div>
```
Comment (stefankorder) [inline components/Entry/index.jsx:104]: Please delete this empty `<div>`.

**Review by stefankorder — CHANGES_REQUESTED** [review]
There are a few changes that have not yet been implemented. And you checks didn't pass.

**Comment by SarahWett** [comment]
> There are a few changes that have not yet been implemented. And you checks didn't pass.

All Changes (and more) are done now. Can you figure out why deployment still fails ? 
No Errors or Warnings inside the localhost environment anymore

File: components/Entry/index.jsx (hunk 1 above)
Comment (SarahWett) [reply components/Entry/index.jsx]: Deleted

File: components/Entry/index.jsx (hunk 2 above)
Comment (SarahWett) [reply components/Entry/index.jsx]: Deleted

File: components/Entry/index.jsx (hunk 3 above)
Comment (SarahWett) [reply components/Entry/index.jsx]: Deleted although we're told to keep comments about where the code os from

File: pages/dashboard/AllEntries/index.js (hunk 12 above)
Comment (SarahWett) [reply pages/dashboard/AllEntries/index.js]: Deleted

File: components/Form/index.js (hunk 14 above)
Comment (SarahWett) [reply components/Form/index.js:126]: Deleted

File: components/Entry/index.jsx (hunk 15 above)
Comment (SarahWett) [reply components/Entry/index.jsx]: Checks pass. Everything already is deleted as you recommended!

**Review by stefankorder — APPROVED** [review]
👍🏻

**Pull request merged**